src/main/java/com/amazon/opendistroforelasticsearch/sql/metrics/MetricName.java

```diff
@@ -0,0 +1,51 @@
+/*
+ *   Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License").
+ *   You may not use this file except in compliance with the License.
+ *   A copy of the License is located at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   or in the "license" file accompanying this file. This file is distributed
+ *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *   express or implied. See the License for the specific language governing
+ *   permissions and limitations under the License.
+ */
+
+package com.amazon.opendistroforelasticsearch.sql.metrics;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public enum MetricName {
+
+    REQ_TOTAL("request_total"),
+    REQ_COUNT_TOTAL("request_count"),
+    FAILED_REQ_COUNT_SYS("failed_request_count_syserr"),
+    FAILED_REQ_COUNT_CUS("failed_request_count_cuserr"),
+    FAILED_REQ_COUNT_CB("failed_request_count_cb"),
+    CIRCUIT_BREAKER("circuit_breaker"),
+    DEFAULT("default");
```

Comment:
what is this one for?

Reply:
FAILED_REQ_COUNT_CUS is for counting request failures by customer error, like SQL syntax error etc.
FAILED_REQ_COUNT_CB is for counting request failures by circuit breaker.
CIRCUIT_BREAKER is current status of circuit breaker, i.e. circuit is broken or not.
DEFAULT is a default metric, could be used in some scenarios in the future.

Reply:
I think it's better to get rid of default, especially that you have a default case in your `MetricFactory::createMetric` anyway.

Reply:
I think these two default are two different concepts. One is for user to use intentionally another one is for some unexpected scenarios.